File: Model.Gui/ModelGui/MainWindow/Model/Environment.cs
namespace Model.Gui
{
    public class Environment
    {
        public Rgba SkyColor { get; set; }
        public Shading Shading { get; set; } = new Shading();
        public Camera Camera { get; set; } = new Camera();

        public Environment Clone() 
        {
            return new Environment() 
            {
                SkyColor = SkyColor,
                Shading = Shading.Clone(),
                Camera = Camera.Clone()
            };
        }
    }
}

File: Model.Render/Render/Scene/Environment.cpp
#pragma once
#include "Environment.h"

Environment::Environment() {
	EnvironmentCamera = 0;
	EnvironmentShading = Shading();
	SkyColor = Rgba();
}

Environment::Environment(Camera* environmentCamera, Shading environmentShading, Rgba skyColor) {
	EnvironmentCamera = environmentCamera;
	EnvironmentShading = environmentShading;
	SkyColor = skyColor;
}